plugins/stackstorm/src/components/ExecutionsTable/ExecutionsTable.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect, useState } from 'react';
+import { Link, Table, TableColumn } from '@backstage/core-components';
+import { useApi, configApiRef, errorApiRef } from '@backstage/core-plugin-api';
+import { Execution, stackStormApiRef } from '../../api';
+import { Status } from './Status';
+import { ExecutionPanel } from './ExecutionPanel';
+
+type DenseTableProps = {
+  executions: Execution[];
+  loading: boolean;
+  page: number;
+  pageSize: number;
+  onPageChange: (page: number) => void;
+  onRowsPerPageChange: (rows: number) => void;
+};
+
+export const DenseTable = ({
+  executions,
+  loading,
+  page,
+  pageSize,
+  onPageChange,
+  onRowsPerPageChange,
+}: DenseTableProps) => {
+  const config = useApi(configApiRef);
+
+  const columns: TableColumn<Execution>[] = [
+    {
+      title: 'Status',
+      field: 'status',
+      render: e => <Status status={e.status} />,
+    },
+    {
+      title: 'Time',
+      field: 'start_timestamp',
+      render: e => `${new Date(e.start_timestamp).toUTCString()}`,
+    },
+    { title: 'Name', field: 'action.ref' },
+    {
+      title: 'Execution ID',
+      field: 'id',
+      render: e => (
+        <Link
+          to={`${config.getString('stackstorm.webUrl')}/?#/history/${e.id}`}
```

Comment:
A note about these URLs and config reads.

You've declared the `webUrl` as optional. What happens here is that late, deep in rendering code inside the interface, errors will be thrown in unexpected ways at the point of user interaction, if the adopter had forgotten to enter that config value. In some of these cases it'll probably lead to a severely broken user interface.

Would it make sense to

- Make `webUrl` a required config parameter
- Update your `StackstormClient` to instead of the public constructor have a

  ```ts
  static fromConfig(
    config: ConfigApi,
    dependencies: {
      discoveryApi: DiscoveryApi;
      fetchApi: FetchApi;
    }
  ): StackstormClient
  ```

  which throws an exception immediately if the config isn't there
- Add methods that resolve URLs to the client, eg `getExecutionHistoryUrl(id): string` that leverages the config and moves that concern out from the rendering code

Perhaps?

Reply:
Yeah makes sense. Done